### ISSUE TITLE
[Feature Request] Support Migration Testing

### ISSUE THREAD

**Issue by pascaldornfeld** [mobile-dev-inc/maestro#1630]
**Is your feature request related to a problem? Please describe.**
As a tester I want to make sure that the app still works after going from version x to version y of my app. A typical testflow is:
- I install an already published app
- I generate some data to fill the app database
- I update the app to the version I want to test
- I check that the data from the old version is still visible

**Describe the solution you'd like**
Support is needed to install apps during the testflow and including when running in maestro cloud.

**Describe alternatives you've considered**
It is possible to install apks during tests with an appium testsuite and migrations can partly be tested with room integration. 

**Additional context**
none

**Comment by bartekpacia** [comment]
This scenario should be made possible by PR #1854, which aims to resolve issue #1665

**Comment by Fishbowler** [comment]
@bartekpacia This is awaiting the PR to move through - I think the automation has been a little aggressive 😁

**Comment by bartekpacia** [comment]
Yeah, it's weird indeed. I'm gonna monitor the situation, thanks!
